cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/BasicLineageWriter.java

```diff
@@ -66,13 +65,19 @@ public BasicLineageWriter(DatasetFramework datasetFramework, TransactionSystemCl
   @Override
   public void addAccess(ProgramRunId run, DatasetId datasetId, AccessType accessType,
                         @Nullable NamespacedEntityId component) {
+    // Don't record lineage for the lineage dataset itself, otherwise there would be infinite recursion
+    if (getLineageDatasetId().equals(datasetId)) {
```

Comment:
In `LineageWriterDatasetFramework.java`, it seems like we don't record access for the system datasets, though?

Reply:
There is no guarantee that lineage dataset id is in system namespace, right? We shouldn't have assumption in code.

Reply:
fair enough - just like how flow queue tables are in user namespace, I suppose.